Sources/CommonMark/Supporting Types/Children.swift

```diff
@@ -133,279 +128,70 @@ extension ContainerOfBlocks {
      - Returns: `true` if successful, otherwise `false`.
      */
     @discardableResult
-    public func remove(child: Block & Node) -> Bool {
+    public func remove(child: ChildNode) -> Bool {
         guard child.parent == self else { return false }
         cmark_node_unlink(child.cmark_node)
         child.managed = true
         return true
     }
 }
 
-// MARK: -
-
-public protocol ContainerOfInlineElements: Node {}
-
-extension Heading: ContainerOfInlineElements {}
-extension Paragraph: ContainerOfInlineElements {}
-extension HTMLBlock: ContainerOfInlineElements {}
-extension CodeBlock: ContainerOfInlineElements {}
-extension ThematicBreak: ContainerOfInlineElements {}
-extension Strong: ContainerOfInlineElements {}
-extension Emphasis: ContainerOfInlineElements {}
-extension Link: ContainerOfInlineElements {}
 
-extension ContainerOfInlineElements {
-    /// The block's children.
-    public var children: [Inline & Node] {
-        get {
-            return Children(of: self).compactMap { $0 as? Inline & Node }
-        }
-
-        set {
-            for child in children {
-                remove(child: child)
-            }
-
-            for child in newValue {
-                append(child: child)
-            }
+extension Container where ChildNode: Node {
+    public init(children: [ChildNode]) {
+        self.init(cmark_node_new(type(of: self).cmark_node_type))
```

Comment:
In context of https://github.com/SwiftDocOrg/CommonMark/pull/14#issuecomment-622612425 I'd guess this line to be the culprit.

Have you tried replacing it with this?

```swift
self.init(cmark_node_new(Self.cmark_node_type))
```

Reply:
Yes, I've tried already. It will indeed make this Swift compiler crash on Linux go away. Unfortunately, it then runs in _yet another_ Swift compiler crash. All the convenience initializers on the nodes which take children cause the following compiler crash:

```
swift: /home/buildnode/jenkins/workspace/oss-swift-5.2-package-linux-ubuntu-18_04/swift/lib/IRGen/GenPoly.cpp:70: void swift::irgen::reemitAsUnsubstituted(swift::irgen::IRGenFunction &, swift::SILType, swift::SILType, swift::irgen::Explosion &, swift::irgen::Explosion &): Assertion `expectedSchema.size() == cast<LoadableTypeInfo>(substTI).getExplosionSize()' failed.
```